app/cmd/backup.go

```diff
@@ -82,16 +83,53 @@ func getBackupStatus(c *cli.Context, backupID string, replicaAddress string) (*s
 	return backupStatus, nil
 }
 
+func getKnownReplicas(c *cli.Context) (map[string]types.Mode, error) {
+	cli := getCli(c)
+	replicas, err := cli.ReplicaList()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get replica list: %v", err)
+	}
+
+	knownReplicas := make(map[string]types.Mode)
+	for _, replica := range replicas {
+		knownReplicas[replica.Address] = replica.Mode
+	}
+
+	return knownReplicas, nil
+}
+
 func fetchAllBackups(c *cli.Context) error {
 	backupProgressList := make(map[string]*sync.BackupStatusInfo)
 
 	cli := getCli(c)
 	backupReplicaMap, err := cli.BackupReplicaMappingGet()
 	if err != nil {
-		return fmt.Errorf("failed to get list of backupIDs:%v", err)
+		return fmt.Errorf("failed to get list of backupIDs: %v", err)
+	}
+
+	knownReplicas, err := getKnownReplicas(c)
+	if err != nil {
+		return err
 	}
 
 	for backupID, replicaAddress := range backupReplicaMap {
+		// a replica is only valid in Read-Write state.
+		// during a rebuild (Write only) the memory information is lost,
+		// so there is no point in requesting the backup status
+		if mode, known := knownReplicas[replicaAddress]; !known || mode != types.RW {
+			err := cli.BackupReplicaMappingDelete(backupID)
+			if err != nil {
+				return err
+			}
+			backupProgressList[backupID] = &sync.BackupStatusInfo{
+				ReplicaAddress: replicaAddress,
+				State:          "error",
+				Error: fmt.Sprintf("Failed to get backup status on %s for %v: %v",
+					replicaAddress, backupID, "replica is no longer known"),
```

Comment:
Reason: `unknown replica`.

Reply:
Not sure if you wanted reason to be part of the message, so I left it out because of the formatting.